.github/workflows/ci.yml

```diff
@@ -15,25 +15,52 @@ jobs:
     timeout-minutes: 20
 
     outputs:
-      image: ${{ steps.tag.outputs.image }}
+      tag: ${{ steps.setup.outputs.tag }}
 
     steps:
-      - uses: actions/checkout@v2
-
-      - name: build image
-        uses: docker/build-push-action@v1
+      -
+        uses: actions/checkout@v2
+      -
+        name: Set image tag
+        id: setup
+        run: |
+          IMAGE_REF=`echo ${{ github.sha }} | sed 's/^\(.......\).*/\1/'`
+          TAG="sha-${IMAGE_REF}"
+          echo "::set-output name=tag::sha-${IMAGE_REF}"
+      -
+        name: Set up QEMU
+        uses: docker/setup-qemu-action@v1
+      -
+        name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
```

Comment:
This thing is supposed to provide access to the early/still-in-development features of the Docker CLI? Do we really need them? Which experimental features exactly are we using?

Reply:
I just followed the example here: https://github.com/docker/build-push-action/tree/master#git-context

Reply:
This step is required, see comment below.

Reply:
ok